kong/db/schema/topological_sort.lua

```diff
@@ -0,0 +1,106 @@
+local table_insert = table.insert
+
+-- Given an array of schemas, build a table were the keys are schemas and the values are
```

Comment:
This seems like quite a general topological sort module. Perhaps move to utils? (or even it's own luarock at some point?)

Reply:
We have another here:
https://github.com/Kong/kong/blob/master/kong/db/strategies/postgres/connector.lua#L65-L109

Which is about the same as:
https://github.com/bungle/lua-resty-tsort

There is also:
https://github.com/starius/toposort/

Reply:
good to know! in the interest of expediency, let's consider refactoring this later